homeassistant/components/light/lightwave.py

```diff
@@ -0,0 +1,97 @@
+"""
+Implements LightwaveRF lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.lightwave/
+"""
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import CONF_DEVICES, CONF_NAME
+from homeassistant.components.light import (
+    Light, ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, PLATFORM_SCHEMA)
+from homeassistant.components.lightwave import LIGHTWAVE_LINK
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA}
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['lightwave']
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Find and return LightWave lights."""
+    lights = []
+    lwlink = hass.data[LIGHTWAVE_LINK]
+
+    for device_id, device_config in config.get(CONF_DEVICES, {}).items():
+        name = device_config[CONF_NAME]
+        lights.append(LRFLight(name, device_id, lwlink))
+
+    async_add_entities(lights)
+
+
+class LRFLight(Light):
+    """Representation of a LightWaveRF light."""
+
+    def __init__(self, name, device_id, lwlink):
+        """Initialize LRFLight entity."""
+        self._name = name
+        self._device_id = device_id
+        self._state = None
+        self._brightness = 255
+        self._lwlink = lwlink
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def should_poll(self):
+        """No polling needed for a LightWave light."""
+        return False
+
+    @property
+    def name(self):
+        """Lightwave light name."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def is_on(self):
+        """Lightwave light is on state."""
+        return self._state
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the LightWave light on."""
+        self._state = True
+
+        if ATTR_BRIGHTNESS in kwargs:
+            self._brightness = kwargs[ATTR_BRIGHTNESS]
+
+        if self._brightness != 255:
```

Comment:
Is it worth extracting 255 to a constant to give some context? It could also be reused in the constructor.

Reply:
Nah it's ok like this I think.

Reply:
Wouldn't it read better like this?

```python
if self._brightness != MAX_BRIGHTNESS
```

Reply:
const for MAX_BRIGHTNESS added. Probably best if this got moved to const.py or light/__init__.py as lots of components require 255 as MAX_BRIGHTNESS.